add-ground-truth.py

```diff
@@ -0,0 +1,218 @@
+"""
+Convert a recording to "ground truth" format and add it to another recording for benchmarking.
+
+For instance, you can get relatively accurate poses with Orbbec Femto Mega/Bolt device and then compare those with OAK-D poses.
+"""
+
+import json
+import os
+import numpy as np
+
+def synchronizeRecordings(args):
+    import sys
+    import importlib
+
+    sys.argv = [
+        'sync-imus.py',
+        f'{os.path.join(args.ground_truth, "data.jsonl")}',
+        f'{os.path.join(args.data, "data.jsonl")}',
+        '--no_plot'
+    ]
+    offset = importlib.import_module("sync-imus").synchronizeImus()
+    return offset
+
+def computeGroundTruth(args):
+    import spectacularAI
+
+    with open(os.path.join(args.ground_truth, "calibration.json")) as f:
+        calibration = json.load(f)
+        imuToPrimaryCamera = np.array(calibration["cameras"][0]["imuToCamera"])
+
+    def poseToJson(position, orientation):
+        poseJson = {}
+        poseJson["position"] = {
+            "x": position.x,
+            "y": position.y,
+            "z": position.z
+        }
+        if orientation is not None:
+            poseJson["orientation"] = {
+                "x": orientation.x,
+                "y": orientation.y,
+                "z": orientation.z,
+                "w": orientation.w
+            }
+        return poseJson
+
+    def isAlreadyRectified(input_dir):
+        vioConfigYaml = f"{input_dir}/vio_config.yaml"
+        if os.path.exists(vioConfigYaml):
+            with open(vioConfigYaml) as file:
+                for line in file:
+                    if "alreadyRectified" in line:
+                        _, value = line.split(":")
+                        return value.lower().strip() == "true"
+        return False
+
+    visualizer = None
+    groundTruthLines = []
+
+    def onVioOutput(output):
+        if visualizer is not None:
+            visualizer.onVioOutput(output.getCameraPose(0), status=output.status)
+
+    def onMappingOutput(output):
+        if visualizer is not None:
+            visualizer.onMappingOutput(output)
+
+        if output.finalMap:
+            for frameId in output.map.keyFrames:
+                keyFrame = output.map.keyFrames.get(frameId)
+                primary = keyFrame.frameSet.primaryFrame
+                pose = primary.cameraPose.pose
+                imuToWorld = pose.asMatrix() @ imuToPrimaryCamera # imu->world = imu->camera->world = camera->world * imu->camera
+                pose = spectacularAI.Pose.fromMatrix(pose.time, imuToWorld)
+                gtJson = {
+                    "time": pose.time,
+                    "groundTruth": poseToJson(pose.position, None if args.no_orientation else pose.orientation)
+                }
+                groundTruthLines.append(gtJson)
+
+    def parseInputDir(input_dir):
+        device = None
+        metadataJson = f"{input_dir}/metadata.json"
```

Comment:
Probably not portable since you use literal `/` signs at some places. You could specify the argparse argument type `input_dir` with `type=pathlib.Path` and then use `metadataJson = input_dir / "metadata.json"`. (Future tip, no need to fix)

Reply:
Copied from here and it has worked previously on both Linux and Windows: https://github.com/SpectacularAI/sdk/blob/64ff00087ce102d6bcd705b450c0e70cd787cb86/python/cli/process/process.py#L633

But yeah, both places should use `os.path.join` instead (I tried to use it elsewhere in this script).

Reply:
Fixed